chalk-solve/src/recursive/fulfill.rs

```diff
@@ -421,8 +421,9 @@ impl<'s, 'db, I: Interner> Fulfill<'s, 'db, I> {
         // inference as an ambiguous solution.
 
         let interner = self.solver.program.interner();
+        let canonical_subst = self.infer.canonicalize(interner, &subst);
```

Comment:
This can be used in the `else` case below instead of recomputing the substitution.

Reply:
done!